gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClientUtils.java

```diff
@@ -0,0 +1,174 @@
+   * @return             - {@link ConfigKeyPath} for the relative path
+   */
+  public static ConfigKeyPath buildConfigKeyPath(URI configKeyURI, ConfigStore cs){
+    checkMatchingSchemeAndAuthority(configKeyURI, cs);
+    // Example store root is   etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest
+    
+    // configKeyURI is etl-hdfs:///datasets/a1/a2
+    if(configKeyURI.getAuthority()==null){
+      return getConfigKeyPath(configKeyURI.getPath());
+    }
+    // configKeyURI is etl-hdfs://eat1-nertznn01.grid.linkedin.com:9000/user/mitu/HdfsBasedConfigTest/datasets/a1/a2
+    else {
+      URI relative = cs.getStoreURI().relativize(configKeyURI);
+      return getConfigKeyPath(relative.getPath());
+    } 
+  }
+  
+  /**
+   * Build the {@link  ConfigKeyPath} based on the absolute/relative path
+   * @param input - absolute/relative file path
+   * @return      - {@link  ConfigKeyPath} corresponding to the input
+   */
+  public static ConfigKeyPath getConfigKeyPath(String input){
+    ConfigKeyPath result = SingleLinkedListConfigKeyPath.ROOT;
+    String[] paths = input.split("/");
+    for(String p: paths){
+      // in case input start with "/", some elements could be "", which should be skip
+      if(p.equals("")){
+        continue;
+      }
+      result = result.createChild(p);
+    }
+    return result;
+  }
+  
+  /**
+   * 
+   * @param configKeyPath - {@link ConfigKeyPath} object which represent the config node in input {@link ConfigStore}
+   * @param cs            - the corresponding ConfigStore
+   * @return              - the absolute URI which combine the ConfigStore root URI and input ConfigKeyPath
+   */
+  public static URI getAbsoluteURI(ConfigKeyPath configKeyPath, ConfigStore cs){
+    URI storeRoot = cs.getStoreURI();
+    String path = storeRoot.getPath();
+    if(path.endsWith("/")){
+      path = path + configKeyPath.getAbsolutePathString().substring(1);
+    }
+    else{
+      path = path + configKeyPath.getAbsolutePathString();
+    }
+    
+    try {
+      return new URI(storeRoot.getScheme(), storeRoot.getAuthority(), path , null, null);
+    } catch (URISyntaxException e) {
+      // should not come here
+      throw new RuntimeException("Can not build URI based on " + configKeyPath);
+    }
+  }
+  
+  /**
+   * batch process for {@link #getAbsoluteURI(Collection, ConfigStore)}
+   * @param configKeyPaths
+   * @param cs
+   * @return
+   */
+  public static Collection<URI> getAbsoluteURI(Collection<ConfigKeyPath> configKeyPaths, ConfigStore cs){
+    Collection<URI> result = new ArrayList<>();
+    for(ConfigKeyPath p: configKeyPaths){
+      result.add(getAbsoluteURI(p, cs));
+    }
+    
+    return result;
+  }
+  
+  private static void checkMatchingSchemeAndAuthority(URI configKeyURI, ConfigStore cs){
+    if(configKeyURI == null || cs == null){
+      throw new IllegalArgumentException("input can not be null");
```

Comment:
nit: I suggest that all checks are either Preconditions or all throw IllegalArgumentException.